torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -787,6 +786,17 @@ def test_duplicate_name(self):
                 rpc_backend_options=self.rpc_backend_options,
             )
 
+    @dist_init(setup_rpc=False)
+    def test_duplicate_name_2(self):
+        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+            rpc.init_rpc(
+                name=worker_name(self.rank % 2),
```

Comment:
nit: Would an error not be raised if world_size <= 2? Maybe it can be `self.rank % (self.world_size - 1)` if that is the case